orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/kubernetes/KubernetesContainerFinder.groovy

```diff
@@ -63,6 +63,15 @@ class KubernetesContainerFinder {
 
     containers.forEach { container ->
       if (container.imageDescription.fromContext) {
+	//  If there are parallel Find images stages , then deploymentDetails would be fetched from it. 
```

Comment:
Can you include a screenshot of your pipeline?

Each `FindImage` stage should write to it's local context  as well as the global context. I could see the global context getting overridden but IMO there is likely value in moving away from that where possible.

For example the `Find Image` stage should have `amiDetails` in it's immediate stage context.

So it's possible we still need a fix here but it'd be nice if we err'd on the side of looking at individual stage contexts in addition to the global context.

cc @robfletcher @duftler